platform/mbed_error.c

```diff
@@ -1,34 +1,8 @@
-/* mbed Microcontroller Library
- * Copyright (c) 2006-2013 ARM Limited
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-#include <stdlib.h>
-#include <stdarg.h>
-#include "device.h"
-#include "platform/mbed_toolchain.h"
-#include "platform/mbed_error.h"
 #include "platform/mbed_interface.h"
-#if DEVICE_STDIO_MESSAGES
-#include <stdio.h>
-#endif
+#include <stdarg.h>
 
-WEAK void error(const char* format, ...) {
-#ifndef NDEBUG
-    va_list arg;
-    va_start(arg, format);
-    mbed_error_vfprintf(format, arg);
-    va_end(arg);
-#endif
-    exit(1);
-}
+// Workaround to provide an externally linkable error function for binary
+// compatiblity with precompiled libraries. Will be gced by compiler if
+// not used at link time. Tiny bit hacky.
+#define static extern
```

Comment:
I dont think we should allow this in our codebase. If linker is not smart enough, complain there.